var/spack/repos/builtin/packages/faiss/package.py

```diff
@@ -56,12 +63,65 @@ class Faiss(AutotoolsPackage, CudaPackage):
     # also, some include paths in gpu/tests/Makefile are missing
     patch("fixes-in-v1.6.3.patch", when="@1.6.3")
 
+    # patch for v1.7.2
+    # a shared object is missing in the python/setup.py
+    #   https://github.com/facebookresearch/faiss/issues/2063
+    #   https://github.com/facebookresearch/faiss/pull/2062
+    # a header is missing in a test file
+    #   https://github.com/facebookresearch/faiss/issues/2300
+    patch("fixes-in-v1.7.2.patch", when="@1.7.2")
+
+    def setup_run_environment(self, env):
+        if "+python" in self.spec:
+            env.prepend_path("PYTHONPATH", python_platlib)
+
+
+class PythonPipBuilder(spack.build_systems.python.PythonPipBuilder):
+    def __init__(self, pkg, build_dirname):
+        spack.build_systems.python.PythonPipBuilder.__init__(self, pkg)
+        self.build_dirname = build_dirname
+
+    @property
+    def build_directory(self):
+        return os.path.join(self.pkg.stage.path, self.build_dirname, "faiss", "python")
+        # return self.pkg.stage.source_path
+
+
+class CMakeBuilder(spack.build_systems.cmake.CMakeBuilder):
+    def cmake_args(self):
+        spec = self.spec
+        args = [
+            self.define_from_variant("BUILD_SHARED_LIBS", "shared"),
+            self.define_from_variant("FAISS_ENABLE_PYTHON", "python"),
+            self.define_from_variant("FAISS_ENABLE_GPU", "cuda"),
+            self.define_from_variant("BUILD_TESTING", "tests"),
+            self.define("FAISS_OPT_LEVEL", "generic"),
+        ]
+        if "+python" in spec:
+            pyexe = spec["python"].command.path
+            args.append(self.define("Python_EXECUTABLE", pyexe))
+
+        if "+cuda" in spec:
+            key = "CMAKE_CUDA_ARCHITECTURES"
+            args.append(self.define_from_variant(key, "cuda_arch"))
+            # args.append(self.define_from_variant(
+            # 'CMAKE_CUDA_STANDARD', 'cudastd'))
+        return args
+
+    def install(self, pkg, spec, prefix):
+        spack.build_systems.cmake.CMakeBuilder.install(self, pkg, spec, prefix)
```

Comment:
Doesn't this:
```suggestion
        super().install(pkg, spec, prefix)
```
work as well?